.github/workflows/ci.yml

```
@@ -25,5 +25,14 @@ jobs:
         version: latest
         skip-cache: true
 
-    - name: Run Go tests
-      run: go test -v -race ./...
+    - name: Run Go tests and generate coverage report
+      run: go test -v -race ./... -coverprofile=coverage.out
+    - name: Upload the coverage report
```

NIT: Line between action job steps would help readability
```suggestion

    - name: Upload the coverage report
```

---

made the suggested changes